packages/fast-components-react-base/src/context-menu-item/context-menu-item.tsx

```diff
@@ -87,30 +87,30 @@ class ContextMenuItem extends Foundation<
      * Handle the keydown event of the item
      */
     private handleMenuItemKeyDown = (e: React.KeyboardEvent<HTMLDivElement>): void => {
-        if (this.props.disabled) {
-            return;
-        }
-
-        if (typeof this.props.onInvoke === "function") {
+        if (typeof this.props.onInvoke === "function" && !this.props.disabled) {
             switch (e.keyCode) {
                 case KeyCodes.enter:
                 case KeyCodes.space:
                     this.props.onInvoke(e, this.props);
                     break;
             }
         }
+
+        if (typeof this.props.onKeyDown === "function") {
+            this.props.onKeyDown(e);
+        }
     };
 
     /**
      * Handle the keydown event of the item
      */
     private handleMenuItemClick = (e: React.MouseEvent<HTMLDivElement>): void => {
-        if (this.props.disabled) {
-            return;
+        if (typeof this.props.onInvoke === "function" && !this.props.disabled) {
+            this.props.onInvoke(e, this.props);
         }
 
-        if (typeof this.props.onInvoke === "function") {
-            this.props.onInvoke(e, this.props);
+        if (typeof this.props.onClick === "function") {

```

Comment:
Do we want to execute this if disabled? I could potentially see a reason, but wonder if it's counterintuitive.

Reply:
yes - this should work as an unhandledProp.